### PR TITLE
Bug fix paged array

### PR DESCRIPTION
**Pull request by kmuseth**
Fixed a bug in util::PagedArray caused by a race condition (due to an incorrect template argument)

### CODE REVIEW

File: openvdb/util/PagedArray.h [hunk 1]
```diff
@@ -185,7 +186,7 @@ namespace util {
 
 template<typename ValueT,
          size_t Log2PageSize = 10UL,
-         template<typename ...> class TableT = std::vector>
+         template<typename ...> class TableT = std::deque>// must allow mutiple threads to call operator[] as long as only one thread calls push_back
```
Comment (pcucka) [inline openvdb/util/PagedArray.h]: If it's not safe to use `std::vector`, would it not be wise to add a static assertion (ideally, one that invokes an `IsThreadsafeRWContainer` type trait)?

Or maybe this class doesn't need to be templated on `TableT` at all?

File: openvdb/util/PagedArray.h [hunk 2]
```diff
@@ -200,9 +201,10 @@ class PagedArray
 
 public:
     using ValueType = ValueT;
+    using Ptr       = SharedPtr<PagedArray>;
 
     /// @brief Default constructor
-    PagedArray() { mSize = 0; }
+    PagedArray() = default;
```
Comment (pcucka) [inline openvdb/util/PagedArray.h]: Revert this change; `tbb::atomic` needs to be initialized by assignment.  (See PR #386.)

File: openvdb/util/PagedArray.h [hunk 3]
```diff
@@ -234,7 +239,7 @@ class PagedArray
     /// @param value value to be added to this PagedArray
     ///
     /// @details Constant time complexity. May allocate a new page.
-    size_t push_back(const ValueType& value)
+    inline size_t push_back(const ValueType& value)
```
Comment (pcucka) [inline openvdb/util/PagedArray.h]: Inline qualifiers are redundant for member functions defined inside the class declaration.

File: openvdb/util/PagedArray.h [hunk 4]
```diff
@@ -510,8 +515,8 @@ class PagedArray
         }
     }
     PageTableT mPageTable;//holds points to allocated pages
-    tbb::atomic<size_t> mSize;// current number of elements in array
-    size_t mCapacity = 0;//capacity of array given the current page count
+    tbb::atomic<size_t> mSize{0};// current number of elements in array
```
Comment (pcucka) [inline openvdb/util/PagedArray.h]: Revert (see PR #386).

File: openvdb/CHANGES [hunk 5]
```diff
@@ -17,6 +17,8 @@ Version 6.2.0 - In development
     - Minor performance improvements to moving and filtering VDB points.
 
     Bug fixes:
+    - Fixed a bug in util::PagedArray that could produce a crash due to a race 
+      condution.
```
[the remark this answers is not in the document]
Comment (kmuseth) [reply openvdb/CHANGES]: done

File: openvdb/util/PagedArray.h (hunk 1 above)
Comment (kmuseth) [reply openvdb/util/PagedArray.h]: removed last template argument

File: openvdb/util/PagedArray.h (hunk 2 above)
Comment (kmuseth) [reply openvdb/util/PagedArray.h]: done

File: openvdb/util/PagedArray.h (hunk 3 above)
Comment (kmuseth) [reply openvdb/util/PagedArray.h]: I'm finding conflicting information on this subject matter. For instance, one of the very first google search results on "C++ inline keyword" includes and a example of an application identical to mine (see code in http://www.cplusplus.com/articles/2LywvCM9/). Other forum discussions seem to suggest that the C++ standard does indeed guarantee that member functions that are defined in the declaration will be treated as inline functions. I'm happy to make the change but I'd feel much better is there was definitive proof that it is indeed redundant. Can you point me to something that settles this?

File: openvdb/util/PagedArray.h (hunk 4 above)
Comment (kmuseth) [reply openvdb/util/PagedArray.h]: done

**Comment by kmuseth** [comment]
addressed review comments

File: openvdb/util/PagedArray.h (hunk 3 above)
Comment (pcucka) [reply openvdb/util/PagedArray.h]: cplusplus.com is notoriously unreliable; use cppreference.com instead.

Here's the cppreference entry on inline functions: https://en.cppreference.com/w/cpp/language/inline.

And here's the C++ standard: http://open-std.org/JTC1/SC22/WG21/docs/papers/2019/n4800.pdf (see line 4 of section 9.1.6, "The inline specifier").

Comment (kmuseth) [reply openvdb/util/PagedArray.h]: cool - second sentence in https://en.cppreference.com/w/cpp/language/inline settled it!

I've already made the code changes
